### PR TITLE
SQIL doc

### PR DESCRIPTION
**Pull request by Will-Nie**
Add Doc for SQIL algorithm

### CODE REVIEW

File: source/hands_on/sqil.rst [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+-------------
+1. SQIL is a **model-free** and **value-based** RL algorithm.
+
+2. SQIL is SQL incorporated with Imitation learning 
+
+3. SQIL supports both **discrete** and **continuous** action spaces.
+
+4. SQIL is an **off-policy** algorithm.
+
+5. In nerveX, SQIL uses **eps-greedy** for exploration.
```
Comment (PaParaZz1) [inline source/hands_on/sqil.rst:20]: rename to DI-engine

File: source/hands_on/sqil.rst [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+               value_gamma: Optional[torch.Tensor] = None,
+               criterion: torch.nn.modules = nn.MSELoss(reduction='none'),
+         ) -> torch.Tensor:
+            """
+            Overview:
+               Multistep (1 step or n step) td_error for q-learning based algorithm
+            Arguments:
+               - data (:obj:`q_nstep_td_data`): the input data, q_nstep_sql_td_data to calculate loss
+               - gamma (:obj:`float`): discount factor
+               - Alpha (:obj:｀float`): A parameter to weight entropy term in a policy equation
```
Comment (PaParaZz1) [inline source/hands_on/sqil.rst:99]: `Alpha` -> `alpha`

File: source/hands_on/sqil.rst [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+         if train_data is not None:
+               learner.train(train_data, collector.envstep)
+
+We also need to modify rewards for new data and demonstation data. Taking the CartPole environment as an example:
+
+ .. code-block:: python
+
+        new_data = collector.collect_data(learner.train_iter, policy_kwargs={'eps': eps})
+        for i in range(len(new_data)):
+            new_data[i].reward = torch.tensor([0.])
```
Comment (PaParaZz1) [inline source/hands_on/sqil.rst:211]: update reward assignment implementation

File: source/hands_on/sqil.rst [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+Regrading the demonstration data, we can leave these rewards as they were.
```
Comment (PaParaZz1) [inline source/hands_on/sqil.rst:213]: add `reference` parts like other doc in `hands on RL`

File: source/hands_on/sqil.rst [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+6. The nerveX implementation of SQIL only supports **discrete** action spaces for now.
+
+7. The advantages for SQIL include more robustness in the face of uncertain dynamics and Naturally incorporation with exploration.
+
+Key Equations or Key Graphs
+---------------------------
+SQL considers a more general maximum entropy policy, such that the optimal policy aims to maximize its entropy at each visited state:
+
+.. image:: images/policy_sqil.png
```
Comment (PaParaZz1) [inline source/hands_on/sqil.rst:30]: align center and scale all the images
